test/index_test.js

```diff
@@ -213,4 +213,36 @@ suite('Indexing', () => {
     
   });
 
+  var insert10Tasks = async function(myns) {
+    var expiry = new Date();
+    expiry.setDate(expiry.getDate() + 10);
+    var res = [];
+    for (var i = 1; i <= 10; i++) {
+      let taskId = slugid.nice();
+      await helper.index.insertTask(myns + '.my-task' + _.toString(i) + '.new' + _.toString(i), {
+        taskId:     taskId,
+        rank:       i,
+        data:       {hello: 'world ' + _.toString(i)},
+        expires:    expiry.toJSON(),
+      });
+      let result = await helper.index.findTask(myns + '.my-task' + _.toString(i) + '.new' + _.toString(i));
+      res.push(result);
+      assert(result.taskId === taskId, 'Wrong taskId');
+    }
+    return res;
+  };
+
+  test('list top-level namespaces (without auth) results', async function() {
+    var myns = slugid.v4();
+    let tasks = await insert10Tasks(myns);
+    let result = await helper.index.listNamespaces(myns, {limit: 1});
+    assert.equal(result.namespaces.length, 1, 'Expected 1 namespace');
+    result.namespaces.forEach(function(obj) {
```

Comment:
This only checks the one returned namespace, and thus never uses the `continuationToken` parameter -- there might still be a bug with that parameter!  (I should know - it took me a week to find a typo in this parameter in the expireEntries function!)

Can you add a bit to this test that calls `listNamespaces` again and again, passing the continuationToken, until it doesn't return a continuationToken?  Count how many times you call it, and assert that it's at least 10.  And keep the check that the resulting namespace matches the regex, too.

Reply:
https://github.com/taskcluster/taskcluster-index/pull/39/files#diff-979945c097652b2859f6887cda2f7178R242

asserted that the `continuationToken` stops returning valid values after atleast 10 calls.